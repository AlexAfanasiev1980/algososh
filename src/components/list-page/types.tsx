export interface IData {
  number?: number | undefined | null;
  index?: number | undefined;
  type: "head" | "tail" | "delHead" | "delTail" | "AddIndex" | "DelIndex" | "";
}