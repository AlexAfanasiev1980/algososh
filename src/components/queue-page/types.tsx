import { ElementStates } from "../../types/element-states";

export interface IColor {
  color?: ElementStates;
  index?: number | undefined;
}

export interface IButtonState {
  add: boolean
  delete: boolean
  clear: boolean
}