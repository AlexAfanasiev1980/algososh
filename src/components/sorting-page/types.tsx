export interface IElements {
  arr: Array<number[]>;
  comparable: Array<number[]>;
  notsorted: Array<number[]>;
  sorted: Array<number[]>;
}

export interface IElement {
  arr: number[];
  comparable: number[];
  notsorted: number[];
  sorted: number[];
}