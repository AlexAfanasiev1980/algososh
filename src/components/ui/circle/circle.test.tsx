import React from "react";
import renderer from "react-test-renderer";
// import { render, screen, fireEvent } from "@testing-library/react";
// import userEvent from "@testing-library/user-event";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Компонент Circle", () => {
  it("Рендер Circle без буквы", () => {
    const circle = renderer.create(<Circle />).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle c буквами", () => {
    const circle = renderer.create(<Circle letter={`rrr`}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle c head", () => {
    const circle = renderer.create(<Circle head={`head`}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle c react-элементом в head", () => {
    const circle = renderer.create(<Circle head={<Circle />}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle c tail", () => {
    const circle = renderer.create(<Circle tail={`tail`}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle c react-элементом в tail", () => {
    const circle = renderer.create(<Circle tail={<Circle />}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle c index", () => {
    const circle = renderer.create(<Circle index={1}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle c пропом isSmall === true", () => {
    const circle = renderer.create(<Circle isSmall={true}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle в состоянии default", () => {
    const circle = renderer.create(<Circle state={ElementStates.Default}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle в состоянии changing", () => {
    const circle = renderer.create(<Circle state={ElementStates.Changing}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });

  it("Рендер Circle в состоянии modified", () => {
    const circle = renderer.create(<Circle state={ElementStates.Modified}/>).toJSON();
    expect(circle).toMatchSnapshot();
  });
});