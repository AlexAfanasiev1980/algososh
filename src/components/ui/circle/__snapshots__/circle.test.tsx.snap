// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Компонент Circle Рендер Circle c head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head string"
  >
    head
  </div>
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle c index 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    1
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle c react-элементом в head 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="cyrcle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle c react-элементом в tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
        data-testid="cyrcle"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Компонент Circle Рендер Circle c tail 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Компонент Circle Рендер Circle c буквами 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      rrr
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle c пропом isSmall === true 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle без буквы 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle в состоянии changing 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle в состоянии default 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Компонент Circle Рендер Circle в состоянии modified 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
    data-testid="cyrcle"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
